pkg/ottl/functions.go

```diff
@@ -203,18 +218,42 @@ func (p *basePath[K]) isComplete() error {
 	return p.nextPath.isComplete()
 }
 
-func newKeys[K any](keys []key) []Key[K] {
+func (p *Parser[K]) newKeys(keys []key) ([]Key[K], error) {
 	if len(keys) == 0 {
-		return nil
+		return nil, nil
 	}
 	ks := make([]Key[K], len(keys))
 	for i := range keys {
+		var par Getter[K]
+		if keys[i].Expression != nil {
+			if keys[i].Expression.Path != nil {
+				arg, err := p.buildGetSetterFromPath(keys[i].Expression.Path)
+				if err != nil {
+					return nil, err
+				}
+				par = arg
+			}
+			if f := keys[i].Expression.Float; f != nil {
+				par = literal[K]{value: *f}
+			}
+			if i := keys[i].Expression.Int; i != nil {
+				par = literal[K]{value: *i}
+			}
```

Comment:
When will these two cases be hit? Would it simplify things if we refactored the grammar to just have keys be a `mathExprLiteral` type instead of a `key` type?